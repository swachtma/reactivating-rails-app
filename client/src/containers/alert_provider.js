import { connect } from 'react-redux';

import { scopes } from '../constants/alerts';
import AlertList from '../components/alert_list';

export default function AlertProvider(scope = scopes[0], WrappedComponent = AlertList){
  let mapStateToProps = (state) => ({
    alerts: state.alerts,
    scope: scope
  });
  
  return connect(mapStateToProps,null)(WrappedComponent);
};