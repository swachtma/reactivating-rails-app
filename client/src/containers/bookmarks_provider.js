import { connect } from 'react-redux';

import BookmarkModal from '../components/bookmark_modal'
import { setBookmarkOffered } from '../actions/bookmarks';
import { routeChapter } from '../actions/routes';

export const calculateDisplayState = (state) => {
  let bm = state.bookmarks;
  let loc = state.location.payload.chapter_id;
  if(bm.prompt_shown || (loc === bm.last_read.chapter && loc === bm.furthest_read.chapter)){
    return false;
  } else {
    return true;
  }
};

const bookmarksProvider = (WrappedComponent = BookmarkModal) => {
  let mapStateToProps = (state) => ({
    displayState: calculateDisplayState(state),
    last_read: state.bookmarks.last_read,
    furthest_read: state.bookmarks.furthest_read
  });
  
  let mapDispatchToProps = (dispatch) => ({
    dispatchSetBookmarkOffered: (bool) => dispatch(setBookmarkOffered(bool)),
    dispatchRoutechapter: (chapter_id) => dispatch(routeChapter(chapter_id))
  });
  
  return connect(mapStateToProps,mapDispatchToProps)(WrappedComponent);
};

export const ConnectedBookmarkModal = bookmarksProvider();
export default bookmarksProvider;