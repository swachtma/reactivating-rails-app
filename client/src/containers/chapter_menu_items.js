import { connect } from 'react-redux';

import ChapterMenuItems from '../components/chapter_menu_items';
import { routeChapter } from '../actions/routes';

const mapStateToProps = (state) => ({
  active_chapter: state.chapters.find(
   (chapter) => chapter.id === state.settings.active_chapter_id
  ),
  chapters: state.chapters
});

const mapDispatchToProps = (dispatch) => ({
  dispatchRouteChapter: (event,target) => dispatch(routeChapter(target.value))
});

export default connect(mapStateToProps,mapDispatchToProps)(ChapterMenuItems);