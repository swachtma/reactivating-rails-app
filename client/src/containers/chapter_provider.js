import { connect } from 'react-redux';
import React, { Component } from 'react';

import ChapterFooter from '../components/chapter_footer';
import NavigationMenus from '../components/navigation_menus';
import { routeChapter } from '../actions/routes';

export const connectToChapters = (WrappedComponent) => 
  connect(mapStateToProps,mapDispatchToProps)(chapterProvider(WrappedComponent));


export const chapterProvider = (WrappedComponent) => class ChapterFilter extends Component {
  static displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
  
  render(){
    const { active_chapter_id, chapters, ...otherProps } = this.props;
    const active_chapter = chapters.find(c => c.id === active_chapter_id);
    const next_chapter = chapters.find(c => c.id === active_chapter_id + 1);
    
    const cleaned_chapters = chapters.map( c =>{
      c.title = c.title.replace(/^chapter\s?/i,"");
      return c;
    });
    
    return  chapters && chapters.length ? 
      <WrappedComponent active_chapter={active_chapter || {id: -1, title: ""}}
        next_chapter={next_chapter} chapters={cleaned_chapters} {...otherProps} /> : null;
  }
};

const mapStateToProps = (state) => ({
  chapters: state.chapters,
  active_chapter_id: state.settings.active_chapter_id
});

const mapDispatchToProps = (dispatch) => ({
  dispatchRouteChapter: (event,target) => dispatch(routeChapter(target.value))
});

// EXPORTABLE CONNECTED COMPONENTS
export const ConnectedChapterFooterLink = connectToChapters(ChapterFooter);
export const ConnectedNavigationMenus = connectToChapters(NavigationMenus);