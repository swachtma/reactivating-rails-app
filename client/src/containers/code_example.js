import React, { Component } from 'react';
import { shape, number, string } from 'prop-types';
import clipboardJS from 'clipboard';

import CodeFence from './lib/code_fence';
import CodeBlock from '../components/nodes/code_block';


class CodeExample extends Component {
  static propTypes = {
    node: shape({ id: number, content: string }).isRequired
  };
  
  constructor(props){
    super(props);
    this.code_fence = new CodeFence(props.node);
    this.section_refs = [];
  }
  
  componentDidMount(){
    if(this.triggerCopy){
      let button = this.triggerCopy.querySelector("button.copy");
      return new clipboardJS(button, {
        text: () => this.code_fence.copy
      });
    }
  }

  collapseSections = (bool) => {
    this.section_refs.forEach((ref) => {
      if(ref){ ref.setCollapse(bool) }
    });
  }
  
  addSectionRef = s => this.section_refs.push(s);
  
  attachCopyTrigger = (trigger) => this.triggerCopy = trigger;
  
  render() {
    let node = {
      block_type: this.code_fence.block_type,
      block_path: this.code_fence.block_path,
      sections: this.code_fence.sections
    };
    
    return (
      <CodeBlock 
        node={node} section_refs={this.section_refs}
        collapseSections={this.collapseSections} addSectionRef={this.addSectionRef}
        attachCopyTrigger={this.attachCopyTrigger}
      />
    );
  }
}

export default CodeExample;