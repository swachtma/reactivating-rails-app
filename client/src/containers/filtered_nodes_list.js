import React, { Component } from 'react';
import { connect } from 'react-redux';
import NodesList from '../components/nodes_list';

export const nodesProvider = (WrappedComponent = NodesList) => 
  class extends Component {
    static displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
    
    render(){
      const { nodes, active_chapter_id } = this.props;
      const node_list = nodes.filter(n => n.chapter_id === active_chapter_id);
      return node_list && node_list.length ? <WrappedComponent nodes={node_list} /> : null;
    }
  };

const mapStateToProps = (state) => ({
  nodes: state.nodes,
  active_chapter_id: state.settings.active_chapter_id
});

export const connectToNodes = (WrappedComponent) =>
  connect(mapStateToProps, null)(nodesProvider(WrappedComponent));
  
export default connectToNodes();