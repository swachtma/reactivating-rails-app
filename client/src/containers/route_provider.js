import { connect } from 'react-redux';
import RouterSwitch from '../components/router_switch';

import GithubIcon from '../components/github_icon';

const routerProvider = (WrappedComponent = RouterSwitch) => {
  let mapStateToProps = (state) => ({
    location: state.location
  });
  
  return connect(mapStateToProps,null)(WrappedComponent);
};

export default routerProvider;

// EXPORTABLE CONNECTED COMPONENTS
export const ConnectedGitHubIcon = routerProvider(GithubIcon);