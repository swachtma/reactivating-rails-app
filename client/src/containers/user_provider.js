import { connect } from 'react-redux';

import UserBadge from '../components/user_badge';
import SideBarSignout from '../components/side_bar_signout';
import { signalSignout } from '../actions/user';

 const connectToUser = (WrappedComponent) => {
   return connect(mapStateToProps,mapDispatchToProps)(WrappedComponent);
};

const mapStateToProps = (state) => ({
  username: state.user.username,
  avatar: state.user.avatar
});

const mapDispatchToProps = (dispatch) => ({
  dispatchSignalSignout: (e) => dispatch(signalSignout())
});

// EXPORTABLE CONNECTED COMPONENTS
export const ConnectedUserBadge = connectToUser(UserBadge);
export const ConnectedSidebarSignOut = connectToUser(SideBarSignout);