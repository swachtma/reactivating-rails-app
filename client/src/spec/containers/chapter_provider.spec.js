/* global expect */
import React, { Component } from 'react';
import { shallow } from 'enzyme';

import config from '../../test_helper';
import { chapterProvider } from '../../containers/chapter_provider';

let defaultProps = {
  active_chapter_id: 1,
  chapters: [
    {id: 1, title: "Chapter 1: Stuff"},
    {id: 2, title: "Chapter 2: Things"},  
    {id: 3, title: "Chapter 3: Stuff & Things"}
  ]
};

class MockComponent extends Component {}
const MockContainer = chapterProvider(MockComponent);

describe("chapterProvider", ()=>{
  it("selects active and next chapter", ()=>{
    let w = shallow(<MockContainer {...defaultProps}/>);
    expect(w.find("MockComponent").props().active_chapter.id).toBe(1);
    expect(w.find("MockComponent").props().next_chapter.id).toBe(2);
  });
  
  it("passes cleaned titles to WrappedComponent", ()=>{
    let w = shallow(<MockContainer {...defaultProps}/>);
    expect(w.find("MockComponent").props().chapters[0].title).toBe("1: Stuff");
  });
  
  it("render nothing if no nodes match filter", () => {
    let w = shallow(<MockContainer chapters={[]} active_chapter_id={1} />);
    expect(w).toMatchSnapshot();
  });
});