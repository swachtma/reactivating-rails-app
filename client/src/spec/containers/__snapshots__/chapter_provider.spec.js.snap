// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chapterProvider render nothing if no nodes match filter 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MockComponent
    active_chapter_id={1}
    chapters={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
